bt_ddos_shield/dns_manager.py

```diff
@@ -0,0 +1,52 @@
+from abc import ABC, abstractmethod
+
+from bt_ddos_shield.domain import Domain
+
+
+class DNSManager(ABC):
+    """
+    Abstract base class for manager handling public IP/domain addresses assigned to validators.
+    """
+
+    def __init__(self):
```

Comment:
Probably yes (also thought about it), but this adress is implementation dependant, so I wasn't sure if it should be in base abstract class. If it would be Domain, then it fits here, but maybe it will be some server ID (or anything else) used in some implementations. And then maybe it should be passed in constructor of this implementation?